src/openforms/forms/api/viewsets.py

```diff
@@ -613,18 +625,42 @@ def logic_rules_list(self, request, *args, **kwargs):
         summary=_("JSON schema"),
         description=_("Generate the JSON schema for a form."),
         parameters=[UUID_OR_SLUG_PARAMETER],
-        request=None,
-        responses={status.HTTP_200_OK: OpenApiTypes.OBJECT},
+        request=FormJsonSchemaOptionsSerializer,
+        responses={
+            status.HTTP_200_OK: OpenApiTypes.OBJECT,
+            status.HTTP_400_BAD_REQUEST: ExceptionSerializer,
+        },
+    )
+    @action(
+        detail=True,
+        methods=["get"],
+        permission_classes=(permissions.IsAdminUser,),
+        renderer_classes=(FormJSONSchemaRenderer,),
     )
-    @action(detail=True, methods=["get"], permission_classes=(permissions.IsAdminUser,))
     def json_schema(self, request, *args, **kwargs):
+        serializer = FormJsonSchemaOptionsSerializer(data=request.GET)
+        serializer.is_valid(raise_exception=True)
+
         form = self.get_object()
 
-        schema = generate_json_schema(
-            form,
-            limit_to_variables=form.formvariable_set.values_list("key", flat=True),
+        # Note that a backend is unique by its key and corresponding form
+        backend = get_object_or_404(
+            FormRegistrationBackend,
+            key=serializer.validated_data["registration_backend_key"],
+            form=form.id,
         )
 
+        try:
+            schema = generate_json_schema(
+                form=form,
+                limit_to_variables=form.formvariable_set.values_list("key", flat=True),
+                backend_id=backend.backend,
+                backend_options=backend.options,
+            )
+        except InvalidBackendIdError as exc:
+            logger.error("invalid_backend_id", exc_info=exc)
+            return HttpResponseBadRequest("Invalid backend id")
```

Comment:
is this something that can be validated in the `FormJsonSchemaOptionsSerializer`?

You could look up the form before instantiating the serializer and then pass it in the context:

```py
FormJsonSchemaOptionsSerializer(data=..., context={"form": form})
```